massa-protocol-worker/src/handlers/block_handler/messages.rs

```diff
@@ -355,3 +355,160 @@ impl Deserializer<BlockMessage> for BlockMessageDeserializer {
         .parse(buffer)
     }
 }
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use std::str::FromStr;
+
+    use massa_models::{block_id::BlockId, operation::OperationId};
+    use massa_serialization::{DeserializeError, Deserializer, Serializer};
+
+    #[test]
+    fn test_lower_limit_message() {
+        let message = super::BlockMessage::DataRequest {
+            block_id: BlockId::from_str("B12DvrcQkzF1Wi8BVoNfc4n93CD3E2qhCNe7nVhnEQGWHZ24fEmg")
+                .unwrap(),
+            block_info: super::AskForBlockInfo::Operations(vec![]),
+        };
+        let mut buffer = Vec::new();
+        let serializer = super::BlockMessageSerializer::new();
+        serializer.serialize(&message, &mut buffer).unwrap();
+        let deserializer =
+            super::BlockMessageDeserializer::new(super::BlockMessageDeserializerArgs {
+                thread_count: 1,
+                endorsement_count: 1,
+                max_operations_per_block: 1,
+                max_datastore_value_length: 1,
+                max_function_name_length: 1,
+                max_parameters_size: 1,
+                max_op_datastore_entry_count: 1,
+                max_op_datastore_key_length: 1,
+                max_op_datastore_value_length: 1,
+                max_denunciations_in_block_header: 1,
+                last_start_period: None,
+            });
+        let (rest, deserialized_message) = deserializer
+            .deserialize::<DeserializeError>(&buffer)
+            .unwrap();
+        assert!(rest.is_empty());
+        match (deserialized_message, message) {
+            (
+                super::BlockMessage::DataRequest {
+                    block_id: block_id1,
+                    block_info: block_info1,
+                },
+                super::BlockMessage::DataRequest {
+                    block_id: block_id2,
+                    block_info: block_info2,
+                },
+            ) => {
+                assert_eq!(block_id1, block_id2);
+                assert_eq!(block_info1, block_info2);
+            }
+            _ => panic!("Wrong message type"),
+        }
+        let message2 = super::BlockMessage::DataResponse {
+            block_id: BlockId::from_str("B12DvrcQkzF1Wi8BVoNfc4n93CD3E2qhCNe7nVhnEQGWHZ24fEmg")
+                .unwrap(),
+            block_info: super::BlockInfoReply::Operations(vec![]),
+        };
+        let mut buffer2 = Vec::new();
+        serializer.serialize(&message2, &mut buffer2).unwrap();
+        let (rest2, deserialized_message2) = deserializer
+            .deserialize::<DeserializeError>(&buffer2)
+            .unwrap();
+        assert_eq!(rest2.len(), 0);
+        match (deserialized_message2, message2) {
```

Comment:
Dumb question but why assert_eq! is not used here?

Reply:
assert!(rest2.is_empty()) should be use here

Reply:
It's not a dumb question, it's me the dumb man 😆